src/main/scala/org/broadinstitute/dsde/firecloud/service/TrialService.scala

```diff
@@ -177,36 +179,67 @@ final class TrialService
-              )
-              thurloeDao.saveTrialStatus(userInfo.id, userInfo, enrolledStatus) map { _ =>
-                // TODO: add user to free-trial billing project / create said project if necessary
-                RequestComplete(NoContent)
-              }
+    thurloeDao.getTrialStatus(userInfo.id, userInfo) flatMap {
+      // can't determine the user's trial status; don't enroll
+      case None => Future(RequestCompleteWithErrorReport(BadRequest, "You are not eligible for a free trial. (Error 10)"))
+      case Some(status) =>
+        status.state match {
+          // user already enrolled; don't re-enroll
+          case Some(TrialStates.Enrolled) => Future(RequestCompleteWithErrorReport(BadRequest, "You are already enrolled in a free trial. (Error 20)"))
+          // user enabled (eligible) for trial, enroll!
+          case Some(TrialStates.Enabled) => enrollUserInternal(userInfo, status)
+          // user in some other state; don't enroll
+          case Some(TrialStates.Disabled) => Future(RequestCompleteWithErrorReport(BadRequest, "You are not eligible for a free trial. (Error 30)"))
+          case Some(TrialStates.Terminated) => Future(RequestCompleteWithErrorReport(BadRequest, "You are not eligible for a free trial. (Error 40)"))
+          case None => Future(RequestCompleteWithErrorReport(BadRequest, "You are not eligible for a free trial. (Error 50)"))
+        }
+    }
+  }
+
+  private def enrollUserInternal(userInfo: UserInfo, status: UserTrialStatus): Future[PerRequestMessage] = {
+
+    val projectId = "fccredits-hafnium-peach-3794" // TODO: Make this status.billingProjectName when GAWB-2911 lands
+    val saToken: WithAccessToken = AccessToken(OAuth2BearerToken(googleDAO.getTrialBillingManagerAccessToken))
+
+    // 1. Check that the assigned Billing Project exists and contains exactly one member, the SA we used to create it
+    rawlsDAO.getProjectMembers(projectId)(saToken) flatMap { members: Seq[RawlsBillingProjectMember] =>
+      if (members.map(_.email.value) != Seq(googleDAO.getTrialBillingManagerEmail)) {
+        Future(RequestCompleteWithErrorReport(InternalServerError, "We could not process your enrollment. (Error 60 - billing project exists but cannot be used)"))
+      } else {
+        // 2. Add the user as Owner to the assigned Billing Project
+        rawlsDAO.addUserToBillingProject(projectId, ProjectRoles.Owner, userInfo.userEmail)(userToken = saToken) flatMap { _ =>
+          // 3. Update the user's Thurloe profile to indicate Enrolled status
+          thurloeDao.saveTrialStatus(userInfo.id, userInfo, enrolledStatusFromStatus(status)) map {
+            case Success(_) => RequestComplete(NoContent)
+            case Failure(e) => {
+              // TODO: Remove user from billing project on failure? Is this even likely?
+              RequestCompleteWithErrorReport(InternalServerError, s"We could not process your enrollment. (Error 70 - ${e.getMessage})")
```

Comment:
yes, ideally, remove the user from the project at this point, leave the user in a clean `Enabled` state so they can try again.